src/models/Queryable.js

```diff
@@ -0,0 +1,68 @@
+import URI from 'urijs';
+import _get from 'lodash.get'
+
+export default class Queryable {
+  constructor (id, json) {
+    this.id = id
+    this._rawJson = json
+    this.usableDefinition = null
+  }
+
+  get field () {
+    return this.id
+  }
+
+  get _hasDetails () {
+    if ('type' in this._rawJson) return true
+    if ('$ref' in this._rawJson) return false
+    return true
+  }
+
+  get uiType () {
+    if (this.usableDefinition === null) return null
+    if (this.usableDefinition.enum) return 'selectField'
+    if (this.usableDefinition.type === 'string') return 'textField'
+    if (this.usableDefinition.type === 'number' || this.usableDefinition.type === 'integer') {
+      // if ('minimum' in this.usableDefinition && 'maximum' in this.usableDefinition) return 'rangeField'
+      return 'numberField'
+    }
+    return null
+  }
+
+  get operatorOptions () {
+    if (this.uiType === null) return null
+    if (this.uiType === 'selectField') {
+      return ["=", "<>"]
+    }
+    if (this.uiType === 'textField') {
+      return ["=", "<>", 'LIKE']
+    }
+    if (this.uiType === 'rangeField') {
+      return ['>', ">=", "<", "<="]
+
+    }
+    if (this.uiType === 'numberField') {
+      return ['>', ">=", "<", "<="]
+    }
+    return null
+  }
+
+  async init () {
+    if (!this._hasDetails) {
+      this.usableDefinition = await this.getDefinitionFromReference()
+    } else {
+      this.usableDefinition = this._rawJson
+    }
+  }
+
+  async getDefinitionFromReference () {
+    const response = await fetch(this._rawJson.$ref)
+    if (!response.ok) return
+    const data = await response.json()
+    
+    const uri = new URI(this._rawJson.$ref)
+    const hash = uri.hash()
+    const hashComponents = hash.replace('#/', '').split('/').join('.')
+    return Object.assign(this._rawJson, _get(data, hashComponents))
```

Comment:
As far as I can see this is used only once, but adds (for the functionality given) a relatively large additional dependency (the source has 900 lines). 

If I understand the docs correctly, we can simply do/add the following instead:
```js
function _get(object, path) {
  if (object === null || typeof object !== 'object') {
    return;
  }
  object = object[path[0]];
  if (typeof object !== 'undefined' && path.length >1) {
    return _get(object, path.slice(1));
  }
  return object;
}
```

You'd need to remove the `join('.')` in the line above.

Reply:
Happy to test the proposed.
I generally look at bundleophobia before adding a package to check it's size contributions, [in this case it's 2kb](https://bundlephobia.com/package/lodash.get@4.4.2), which isn't terribly large, but obviously larger than your snippet so if we can trim that down why not :)